lib/tdf3/src/tdf.ts

```diff
@@ -981,27 +982,33 @@ export async function sliceAndDecrypt({
   cryptoService: CryptoService;
   segmentIntegrityAlgorithm: IntegrityAlgorithm;
 }) {
-  for (const index in slice) {
-    const { encryptedOffset, encryptedSegmentSize } = slice[index];
-
-    const offset =
-      slice[0].encryptedOffset === 0 ? encryptedOffset : encryptedOffset % slice[0].encryptedOffset;
-    const encryptedChunk = new Uint8Array(
-      buffer.slice(offset, offset + (encryptedSegmentSize as number))
-    );
+  return Promise.all(
```

Comment:
Unfortunetly it wont change anything. When you do Promise.All on decrypt its not paralelise decryption. I tried it donesnt give any peformnace boost.

https://github.com/opentdf/client-web/pull/256/files#diff-5c6d9ea24cc7a3470e79efacc4f7aed59b657798a7cba23499d0cfd6476b9935R970

here work that im doing for that uses workers. But without limiting promise.all of lot of chunks can freeze ui so i use pLimit.

Could you remove `Promise.all` for now since changes is on the way? and leave just _reject part?

Reply:
Done.